src/udiskslinuxblock.c

```diff
@@ -434,6 +434,34 @@ update_hints (UDisksLinuxBlock  *block,
 }
 
 /* ---------------------------------------------------------------------------------------------------- */
+static void
+process_device_symlinks (const gchar      *device,
+                         UDisksLinuxBlock *block,
+                         gpointer         object,
+                         GList            **ret)
+{
+  const gchar *const *symlinks;
+  guint n;
+
+  if (g_strcmp0 (device, udisks_block_get_device (UDISKS_BLOCK (block))) == 0)
+    {
+      *ret = g_list_prepend (*ret, g_object_ref (object));
+    }
+  else
+    {
+      symlinks = udisks_block_get_symlinks (UDISKS_BLOCK (block));
+      if (symlinks != NULL)
+        {
+          for (n = 0; symlinks[n] != NULL; n++)
+            {
+              if (g_strcmp0 (device, symlinks[n]) == 0)
+                {
+                  *ret = g_list_prepend (*ret, g_object_ref (object));
+                }
+            }
+        }
+    }
```

Comment:
I myself tend to use curly brackets in C even in places where they are not needed. But due to the coding style used in udisks, this is really crazy. I think getting rid of all those extra curly brackets here would make the function much better readable.